src/imitation/algorithms/preference_comparisons.py

```diff
@@ -713,20 +729,15 @@ def variance_estimate(self, rews1, rews2) -> float:
             the variance estimate based on the `uncertainty_on` flag.
         """
         if self.uncertainty_on == "logit":
-            rews1, rews2 = rews1.sum(0), rews2.sum(0)
-            var_estimate = (rews1 - rews2).var().item()
+            returns1, returns2 = rews1.sum(0), rews2.sum(0)
+            var_estimate = (returns1 - returns2).var().item()
         else:  # uncertainty_on is probability or label
-            probs = (
-                self.preference_model.probability(
-                    rews1,
-                    rews2,
-                )
-                .cpu()
-                .numpy()
-            )
+            probs = self.preference_model.probability(rews1, rews2)
+            probs = probs.cpu().numpy()
+            assert probs.shape == (self.preference_model.model.num_members,)
             if self.uncertainty_on == "probability":
                 var_estimate = probs.var()
-            else:  # uncertainty_on is label
+            elif self.uncertainty_on == "label":  # uncertainty_on is label
```

Comment:
Inclined to add a catch all `else: assert False`  though mostly just to make it self-documenting (even without this it'd still error, as `var_estimate` would not be defined, just in a less intuitive way)